sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5663,9 +5767,27 @@ pi_result piextUSMHostAlloc(void **ResultPtr, pi_context Context,
   return PI_SUCCESS;
 }
 
-pi_result piextUSMFree(pi_context Context, void *Ptr) {
+static pi_result piextUSMFreeImpl(pi_context Context, void *Ptr) {
+  pi_platform Plt = Context->Devices[0]->Platform;
+  auto MemAlloc =
+      std::find_if(std::begin(Plt->MemAllocs), std::end(Plt->MemAllocs),
```

Comment:
I think this can get very slow. should we create a map of <Ptr,MemAlloc>?

Reply:
Fixed.